rpc/block_test.go

```diff
@@ -93,50 +93,60 @@ func TestBlockWithTxHashes(t *testing.T) {
-			TransactionHash{TransactionHash: utils.TestHexToFelt(t, "0x4e52da53e23d92d9818908aeb104b007ea24d3cd4a5aa43144d2db1011e314f")},
-			TransactionHash{TransactionHash: utils.TestHexToFelt(t, "0x6cc05f5ab469a3675acb5885c274d5143dca75dd9835c582f59e85ab0642d39")},
-			TransactionHash{TransactionHash: utils.TestHexToFelt(t, "0x561ed983d1d9c37c964a96f80ccaf3de772e2b73106d6f49dd7c3f7ed8483d9")},
-		},
+		Transactions: utils.TestHexArrToFelt(t, []string{
+			"0x40c82f79dd2bc1953fc9b347a3e7ab40fe218ed5740bf4e120f74e8a3c9ac99",
+			"0x28981b14353a28bc46758dff412ac544d16f2ffc8dde31867855592ea054ab1",
+			"0x41176c650076712f1618a141fc1cf9a8c39f0d9548a3458f29cf363310a1e72",
+			"0x43cd66f3ddbfbf681ab99bb57bf9d94c83d6e9b586bdbde78ab2deb0328ebd5",
+			"0x7602cfebe4f3cb3ef4c8b8c6d7dda2efaf4a500723020066f5db50acd5095cd",
+			"0x2612f3f870ee7e7617d4f9efdc41fa8fd571f9720b059b1aa14c1bf15d3a92a",
+			"0x1a7810a6c68adf0621ed384d915409c936efa0c9d436683ea0cf7ea171719b",
+			"0x26683aeef3e9d9bcc1f0d45a5f0b67d0aa1919726524b2a8dc59504dacfd1f4",
+			"0x1d374aa073435cdde1ec1caf972f7c175fd23438bb220848e71720e00fd7474",
+			"0xfc13eabaa2f38981e68bb010370cad7a7d0b65a59101ec816042adca0d6841",
+			"0x672d007224128b99bcc145cd3dbd8930a944b6a5fff5c27e3b158a6ff701509",
+			"0x24795cbca6d2eba941082cea3f686bc86ef27dd46fdf84b32f9ba25bbeddb28",
+			"0x69281a4dd58c260a06b3266554c0cf1a4f19b79d8488efef2a1f003d67506ed",
+			"0x62211cc3c94d612b580eb729410e52277f838f962d91af91fb2b0526704c04d",
+			"0x5e4128b7680db32de4dff7bc57cb11c9f222752b1f875e84b29785b4c284e2a",
+			"0xdb8ad2b7d008fd2ad7fba4315b193032dee85e17346c80276a2e08c7f09f80",
+			"0x67b9541ca879abc29fa24a0fa070285d1899fc044159521c827f6b6aa09bbd6",
+			"0x5d9c0ab1d4ed6e9376c8ab45ee02b25dd0adced12941aafe8ce37369d19d9c2",
+			"0x4e52da53e23d92d9818908aeb104b007ea24d3cd4a5aa43144d2db1011e314f",
+			"0x6cc05f5ab469a3675acb5885c274d5143dca75dd9835c582f59e85ab0642d39",
+			"0x561ed983d1d9c37c964a96f80ccaf3de772e2b73106d6f49dd7c3f7ed8483d9",
+		}),
 	}
 
+	txHashes := utils.TestHexArrToFelt(t, []string{
+		"0x40c82f79dd2bc1953fc9b347a3e7ab40fe218ed5740bf4e120f74e8a3c9ac99",
+		"0x28981b14353a28bc46758dff412ac544d16f2ffc8dde31867855592ea054ab1",
+	})
+
 	testSet := map[string][]testSetType{
-		"mock": {{
-			BlockID: BlockID{Tag: "latest"},
-			ExpectedPendingBlockWithTxHashes: &PendingBlock{
-				ParentHash:       &felt.Zero,
-				Timestamp:        123,
-				SequencerAddress: &felt.Zero,
+		"mock": {
+			{
+				BlockID: BlockID{Tag: "latest"},
+				ExpectedPendingBlockWithTxHashes: &PendingBlockTxHashes{
+					ParentHash:       &felt.Zero,
+					Timestamp:        123,
+					SequencerAddress: &felt.Zero,
+					Transactions:     txHashes,    
+				},
 			},
-		},
 			{
 				BlockID: BlockID{Hash: &felt.Zero},
-				ExpectedBlockWithTxHashes: &Block{
+				ExpectedBlockWithTxHashes: &BlockTxHashes{
 					BlockHeader: BlockHeader{
+						BlockHash:        &felt.Zero,
```

Comment:
Setting the BlockHash to Zero here means that the Block isn't actually checked in the test in lines [194-196](https://github.com/NethermindEth/starknet.go/pull/354/files#diff-6e6821e21ccb8985a9ddf2d9eb0f9a494f729218e6d04a70247837a19431e3a0R194-R196). 

It might be better to set it to some other value (you'll probably have to update the mock function as well)

Reply:
You are right! I added this because of the mock_test but forgot about that condition below